### DISCUSSION TITLE
SSH Key is not working

### DISCUSSION THREAD

**Discussion by jxflx** [community/community#23215] [New to GitHub]
<p>I need help. I’m having problems  accesing with my SSH Key. I’ve done all the steps to authenticate with a SSH Key but it didn’t work:</p>
<ol>
<li>I made a key with ssh-keygen.</li>
<li>open ~/.ssh/id_rsa.pub and copy the key.</li>
<li>Go to account settings on <a href="http://github.com" rel="noopener nofollow ugc">github.com</a>
</li>
<li>Go to SSH Keys</li>
<li>Click on the Add Key button.</li>
<li>paste the key into the key box.</li>
<li>Save the key (enter my github password to verify).</li>
</ol>
<p>I’m running git 2.25.1 over WSL 2 (Ubuntu) on a Windows 10 machine.</p>

**Comment by canuckjacq** [comment]
<p>It looks as though you have successfully added keys to your account so I think that’s sorted.</p>
<p>If you’re having trouble authenticating with your SSH key, we have some steps here for <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/troubleshooting-ssh">troubleshooting SSH</a>.</p>
<p>If that doesn’t work, you can always share what you are trying to do and what kind of errors you’re seeing- that will help folks come up with suggestions.</p>

**Comment by jxflx** [comment]
<p>Even though I have already placed my key correctly, when I try to <code>git push</code> to one of my repositories it keeps asking for my credentials.</p>
<pre><code class="lang-auto">jesus1554@jesus1554:/mnt/c/Users/jesus/prog/sotano$ git push origin publish
Username for 'https://github.com':
</code></pre>
<p>What can I do? The use of github credentials on the terminal is near to be deprecated.</p>

**Comment by byrneh** [comment, marked as answer]
<p><a class="mention" href="/jesus1554">@jesus1554</a>, here are a couple of things to check<br>
Within your local Git have you configured the correct URL for SSH as opposed to HTTPS protocol<br>
SSH: <a href="mailto:git@github.com">git@github.com</a>:jesus1554/sotano.git<br>
HTTPS: <a href="https://github.com/jesus1554/sotano.git" rel="noopener nofollow ugc">https://github.com/jesus1554/sotano.git</a><br>
Some further details here <a href="https://docs.github.com/en/free-pro-team@latest/github/using-git/changing-a-remotes-url" rel="noopener nofollow ugc">changing-a-remotes-url</a></p>
<p>You can still choose to use HTTPS, with a personal access token (instead of a password) if you prefer to having to use SSH.<br>
If you use HTTPS you can create a token with the needed repository permissions, see <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token" rel="noopener nofollow ugc">creating-a-personal-access-token</a></p>
<p>Once you have a token, you can enter it instead of your password when performing Git operations over HTTPS.<br>
For example, on the command line you would enter the following:<br>
$ git clone <a href="https://github.com/username/repo.git" rel="noopener nofollow ugc">https://github.com/username/repo.git</a><br>
Username: your_username<br>
Password: <em>your_token</em></p>

**Comment by jxflx** [comment]
<p>The problem now is resolved.</p>
<p>Thanks!</p>

**Comment by interglobalmedia** [comment]
Hi,

I just encountered the same problem from nowhere. The only thing is by accident I typed git push origin mains instead of main, and then when I did correctly, this happened. Please advise! Thanks. And I got the following back in Terminal (excluding sensitive info):

 WARNING: REMOTE HOST IDENTIFICATION HAS CHANGED!     @
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
IT IS POSSIBLE THAT SOMEONE IS DOING SOMETHING NASTY!
Someone could be eavesdropping on you right now (man-in-the-middle attack)!
It is also possible that a host key has just been changed.

The fingerprint for the RSA key sent by the remote host is
XXXXXXXXXXXXXXXXXXXX

Please contact your system administrator.
Add correct host key in /Users/xxxxxxxx/.ssh/known_hosts to get rid of this message.
Offending RSA key in /Users/xxxxxxxx/.ssh/known_hosts:3
Host key for github.com has changed and you have requested strict checking.
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Reply by airtower-luna** [reply]
They changed the SSH RSA key this morning after the private key was accidentally published, see the announcement here: https://github.blog/2023-03-23-we-updated-our-rsa-ssh-host-key/

The fingerprint published in the docs has been updated too: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints

The announcement describes how to update your `known_hosts` file. Make sure to check the fingerprint before accepting the new key. :slightly_smiling_face:

**Comment by interglobalmedia** [comment]
Just found that out.. They are so "nice" to let everyone know. i had to accidentally come across an article that was just published about this to find out AFTER having tried a million things to fix it. Still can't gain access to my repos which means can do NO WORK. I will look a the links you have provided. But I have to walk away from this for a bit. I won't share the words I am feeling right now about (Microsoft) Github.

**Comment by interglobalmedia** [comment]
I read the post [We updated our RSA SSH host key](https://github.blog/2023-03-23-we-updated-our-rsa-ssh-host-key) that you shared here.,If i understand them correctly, they are stating that we would have to use THEIR key pair???? ]Because i tried  generating a new key pair of my own with the new SSH protocol, and it did not work. Someone has to clarify all this. Something is very fishy about all this. Thanks so much for sharing your information with me. I really appreciate it!

**Reply by airtower-luna** [reply]
SSH uses mutual authentication: You authenticate to the server using your key pair, and the server authenticates to you using its own key pair (that's the one that got compromised). The authentication from the server is critical so you can be sure you're connecting to the right server.

See also: https://en.wikipedia.org/wiki/Public-key_cryptography

**Reply by interglobalmedia** [reply]
Hi @airtower-luna I actually am connected now. I even upgraded to the ed25519 SSH. However, when I check to see whether the fingerprint which is returned to me matches the one that Github has in their post, it does not match it. I have my own. I was not able to add the new key using the curl command. I had to add it manually. What does that mean? I can connect. No problem there so far.

**Comment by interglobalmedia** [comment]
I then tested my SSH connection, and also verified my public key, and everything seems good to go as expected. This was some (long)journey to figure out exactly if one is set up properly for SSH, and if the authentication process is valid. And sometimes confusing if you follow exactly the steps word by word. Thanks again for your help! Much appreciated.

**Comment by awebartisan** [comment]
@interglobalmedia I am going through this right now. I was using ED25519 from the beginning so I shouldn't have been affected at all but I did.

How did you manage to solve it?

**Comment by ctjlewis** [comment]
Why is this so unstable? You have the key, I have the key, the signature is good, what is the issue?

**Comment by nikgautamgithub** [comment]
Why did it suddenly decided not to work??

**Reply by Davidebyzero** [reply]
> Why did it suddenly decided not to work??

I had this issue too, 3 hours ago, whereas it had worked 11 hours ago. I use TortoiseGit, which uses TortoiseGitPlink and Pageant (based on PuTTY). The thing is, I also use PuTTY itself.

I got it working. I believe there were two issues going on for me. One was that GitHub probably updated their public key, and the old one was being cached in `HKEY_CURRENT_USER\Software\SimonTatham\PuTTY\SshHostKeys` without being updated. The other was that TortoiseGitPlink was using only my `PublicKeyFile` set in `HKEY_CURRENT_USER\Software\SimonTatham\PuTTY\Sessions\Default%20Settings`, and ignoring my `puttykeyfile` setting in `.git/config`.

If you are using this software, I'd suggest doing what I did – delete the github.com key from `SshHostKeys` in the registry, then connect to `git@github.com` with PuTTY.exe itself, pointing it to your SSH key (or just run `TortoiseGitPlink.exe -v -i "yourkeyfile.ppk" git@github.com`). And erase the string in the aforementioned `PublicKeyFile` registry setting if you don't want it to override your per-project TortoiseGit setting.
